src/screens/DiscoveryScreenV2/fragment/UsersFragment.js

```diff
@@ -127,7 +127,7 @@ const UsersFragment = ({
         user_id: myId,
         other_id: item.user_id,
         username: item.username,
-        following: item.following
+        following: followedUserIds.find((userId) => userId === item.user_id) ? true : item.following
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The change made here is a good improvement. It checks if the `item.user_id` exists in the `followedUserIds` array and assigns `true` to `following` if it does, otherwise it retains the original value of `item.following`. This ensures that the `following` status is accurately reflected based on the current state of `followedUserIds`.

However, using `Array.prototype.find()` might not be the most efficient way to check for existence in an array, especially if `followedUserIds` can be large. The `find()` method will traverse the entire array until it finds a match, which could potentially be at the very end.

A more performant approach would be to use a `Set` for `followedUserIds`, and then use the `has()` method, which has an average time complexity of O(1), making it faster for larger data sets.

```diff
- following: followedUserIds.find((userId) => userId === item.user_id) ? true : item.following
+ following: followedUserIds.has(item.user_id) ? true : item.following
```

This change assumes that you convert `followedUserIds` to a `Set` when it is initially created or updated. If this isn't feasible due to other parts of your code relying on `followedUserIds` being an array, then your current implementation with `find()` is fine as it is.


<!-- This is an auto-generated comment by OSS CodeRabbit -->